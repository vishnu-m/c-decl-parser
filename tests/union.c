/* Ordinary union */
union check {
        const int x, y;
};

/* Unnamed union */
union {
        const int a;
}u1;

/* Union containing Struct */

union un2 {
        struct st1 {
                double b;
        };
        char *s;
};

/* Unnamed union enclosing unnamed struct containing bit fields */
union {                   
        struct {
                unsigned int icon : 8;
                unsigned color : 4;
        } window1;
        int screenval;
}screen[25][80];