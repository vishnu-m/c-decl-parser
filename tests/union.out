UnionDecl	"check"		Parent = "union.c"
FieldDecl	"x"		type = "const int"		Parent = "check"
FieldDecl	"y"		type = "const int"		Parent = "check"
UnionDecl	""		Parent = "union.c"
FieldDecl	"a"		type = "const int"		Parent = ""
VarDecl		"u1"		storage class specifier = ""		type = "union (anonymous union at union.c:7:1)"
UnionDecl	"un2"		Parent = "union.c"
StructDecl	"st1"		Parent = "un2"
FieldDecl	"b"		type = "double"		Parent = "st1"
FieldDecl	"s"		type = "char *"		Parent = "un2"
UnionDecl	""		Parent = "union.c"
StructDecl	""		Parent = ""
FieldDecl	"icon"		type = "unsigned int"		(bitfield = 8)		Parent = ""
FieldDecl	"color"		type = "unsigned int"		(bitfield = 4)		Parent = ""
FieldDecl	"window1"		type = "struct (anonymous struct at union.c:22:9)"		Parent = ""
FieldDecl	"screenval"		type = "int"		Parent = ""
VarDecl		"screen"		storage class specifier = ""		type = "union (anonymous union at union.c:21:1) [25][80]"
